_adr/81/index.adoc

```diff
@@ -19,10 +19,285 @@ NOTE: https://datatracker.ietf.org/doc/html/rfc2119[RFC-2119] keywords are used
+For example a shard on AWS might support a different set of instance types to a shard on Azure, or the same instance types might have different sizes on those clouds.
+One of the purposes of introducing the instance type and size concepts is to provide flexibility in the kinds of service offered in the future.
+
+* The availability of an instance type might be limited to a particular cloud, region or combination of both.
+
+* It is a private concern of the shard exactly how each size in an instance type is provisioned.
+In principle it could use vertical and/or horizontal scaling and/or density > 1, in addition to applying quotas to limit capacity to the advertised limits.
+The fleet manager doesn't know.
+
+=== UI/Fleet manager interface
+
+This ADR does not mandate any particular presentation for how the scale options are presented in the UI.
+In particular while there is only one instance type defined, there is no requirement to display instance types in the UI at all.
+
+The fleet manager SHALL be the source of truth for the instance types and sizes available, and how those depend on cloud and region. 
+The fleet manager SHALL provide an endpoint which the UI MUST use to determine the available instance types and sizes.
+The UI MUST NOT make assumptions about linearity of scaling, instead it uses only the instance types and sizes provided by the endpoint.
+
+==== Instance type discovery
+
+Reusing the "Standard" instance type defined above, let's illustrate how the Fleet Manager could communicate the available kinds of cluster to a UI.
+
+.Sequence diagram of UI/Fleet manager interaction
+image::adr-81-ui-fleet_manager.png[UI/Fleet manager interaction]
+
+After the UI has discovered the available cloud providers and regions it would make another GET request `/api/kafkas_mgmt/v1/instance_types/{cloudId}/{cloudRegion}` to discover the instance types available in this cloud and region.
+An example response JSON is shown below.
+
+[source,json]
+.Example instance type available to a UI from the Fleet Manager's `GET /api/kafkas_mgmt/v1/instance_types/{cloudId}/{cloudRegion}` endpoint
+----
+"instance_types": [ <1>
+    { "id": "standard", <2>
+      "parameters": { <3>
+        "ingressThroughputPerSec" : { "name": "Ingress", "unit": "MiB/s" },
+        "egressThroughputPerSec": { "name": "Egress", "unit": "MiB/s" },
+        "maxPartitions": { "name": "Partitions" },
+        "totalMaxConnections": { "name": "Connections" },
+        "maxConnectionAttemptsPerSec": { "name": "Connection Rate", "unit": "/s" },
+        "maxMessageSize": { "name": "Maximum message size", "unit": "MiB" },
+        "quotaType": { "name": "The kind of quota an instance of this plan consumes" }, <4>
+        "quotaConsumed": { "name": "The amount of quota instance of this plan consumes" }, <5>
+        "capacityConsumed": { "name": "An amount of shard capacity consumed by an instance of this plan" } <6>
```

Comment:
@pmuir @pb82 I think this is what we agreed, right?

I don't _particularly_ like how these three are magically common to all instance types, whereas there's no requirement for other parameters to be, but I'm not sure defining something parallel to parameters is really worth while.

Reply:
Yes, looks good to me. And what we will use in app-iterface: https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/34777/diffs#8b58cbce883c482be4c1ce504427b2d7776e236c_277_256